cmd/kops/toolbox_bundle.go

```diff
@@ -0,0 +1,215 @@
+	if err != nil {
+		return fmt.Errorf("error reading InstanceGroup %q: %v", groupName, err)
+	}
+	if ig == nil {
+		return fmt.Errorf("InstanceGroup %q not found", groupName)
+	}
+
+	builder := bundle.Builder{
+		Clientset: clientset,
+	}
+	bundleData, err := builder.Build(cluster, ig)
+	if err != nil {
+		return fmt.Errorf("error building bundle: %v", err)
+	}
+
+	sshUser := os.Getenv("USER")
+
+	nodeSSH := &kutil.NodeSSH{
+		Hostname: options.Target,
+	}
+	nodeSSH.SSHConfig.HostKeyCallback = ssh.InsecureIgnoreHostKey()
+	nodeSSH.SSHConfig.User = sshUser
+	sshIdentity := filepath.Join(os.Getenv("HOME"), ".ssh", "id_rsa")
+	if err := kutil.AddSSHIdentity(&nodeSSH.SSHConfig, sshIdentity); err != nil {
+		return err
+	}
+
+	sshClient, err := nodeSSH.GetSSHClient()
+	if err != nil {
+		return fmt.Errorf("error getting SSH client: %v", err)
+	}
+
+	if err := runSshCommand(sshClient, "sudo mkdir -p /etc/kubernetes/bootstrap"); err != nil {
+		return err
+	}
+
+	root, err := nodeSSH.Root()
+	if err != nil {
+		return fmt.Errorf("error connecting to nodeSSH: %v", err)
+	}
+	for _, file := range bundleData.Files {
+		sshAcl := &vfs.SSHAcl{
+			Mode: file.Header.FileInfo().Mode(),
+		}
+		p := root.Join("etc", "kubernetes", "bootstrap", file.Header.Name)
+		glog.Infof("writing %s", p)
+		if err := p.WriteFile(file.Data, sshAcl); err != nil {
+			return fmt.Errorf("error writing file %q: %v", file.Header.Name, err)
+		}
+	}
+
+	if err := runSshCommand(sshClient, "sudo /etc/kubernetes/bootstrap/bootstrap.sh"); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func runSshCommand(sshClient *ssh.Client, cmd string) error {
+	s, err := sshClient.NewSession()
+	if err != nil {
+		return fmt.Errorf("error creating ssh session: %v", err)
+	}
+
+	var stdout bytes.Buffer
+	var stderr bytes.Buffer
+
+	s.Stdout = io.MultiWriter(&stdout, os.Stdout)
+	s.Stderr = io.MultiWriter(&stderr, os.Stderr)
+
+	glog.Infof("running %s", cmd)
+	if err := s.Run(cmd); err != nil {
+		return fmt.Errorf("error running %s: %v\nstdout: %s\nstderr: %s", cmd, err, stdout.String(), stderr.String())
+	}
+
+	glog.Infof("stdout: %s", stdout.String())
+	glog.Infof("stderr: %s", stderr.String())
+	return nil
+}
+
+func writeToTar(files []*bundle.DataFile, bundlePath string) error {
+	f, err := os.Create(bundlePath)
+	if err != nil {
+		return fmt.Errorf("error creating output bundle file %q: %v", bundlePath, err)
+	}
+	defer f.Close()
+
+	gw := gzip.NewWriter(f)
+	defer gw.Close()
+	tw := tar.NewWriter(gw)
+	defer tw.Close()
+
+	for _, file := range files {
+		if err := tw.WriteHeader(&file.Header); err != nil {
+			return fmt.Errorf("error writing tar file header: %v", err)
+		}
+
+		if _, err := tw.Write(file.Data); err != nil {
+			return fmt.Errorf("error writing tar file data: %v", err)
+		}
+	}
+
+	return nil
+}
+
+// bazel build //cmd/kops && bazel-bin/cmd/kops/kops toolbox bundle --name ${CLUSTER} ${IGNAME} && scp output.tar.gz ${TARGET}:/tmp/output.tar.gz
```

Comment:
Is this docs?

Reply:
It was more notes about how to use it.  I'll remove it.